src/utils/Draw.ts

```diff
@@ -35,22 +37,14 @@ export const draw = (network: Network, container: HTMLDivElement) => {
 
     createTextElements(labelNodesGroup, nodes);
     createLabels(labelNodesGroup, nodes);
-
     createDetailsButton(zoomLayer);
 
-    labelNodesGroup
-        .selectAll<SVGGElement, DependencyNode>('g')
-        .on('click', (node: DependencyNode) => {
-            LevelStorage.reset();
-            if (node.links.length) {
-                highlight(node, linkElements);
-                zoomToHighLightedNodes();
-            }
-            event.stopPropagation();
-        })
-        .call(handleDrag(simulation));
+    labelNodesGroup.selectAll<SVGGElement, DependencyNode>('g').call(handleDrag(simulation));
 
-    setKeyboardDependencyHighlightHandler();
+    subscribeToHighlight();
```

Comment:
this is called only once, when application is launched right ?

Reply:
Good catch, it is called everytime environment is changed - I will create a function for cleanup and unsubscribe from all events there